subsys/bluetooth/host/classic/l2cap_br.c

```diff
@@ -238,18 +240,71 @@ static uint8_t l2cap_br_get_ident(void)
 	return ident;
 }
 
+static void raise_data_ready(struct bt_l2cap_br_chan *br_chan)
+{
+	if (!atomic_set(&br_chan->_pdu_ready_lock, 1)) {
+		sys_slist_append(&br_chan->chan.conn->upper_data_ready,
+				 &br_chan->_pdu_ready);
+		LOG_DBG("data ready raised");
+	} else {
+		LOG_DBG("data ready already");
+	}
+
+	bt_conn_data_ready(br_chan->chan.conn);
+}
+
+static void lower_data_ready(struct bt_l2cap_br_chan *br_chan)
+{
+	struct bt_conn *conn = br_chan->chan.conn;
+	sys_snode_t *s = sys_slist_get(&conn->upper_data_ready);
+
+	__ASSERT_NO_MSG(s == &br_chan->_pdu_ready);
+	(void)s;
+
+	atomic_t old = atomic_set(&br_chan->_pdu_ready_lock, 0);
+
+	__ASSERT_NO_MSG(old);
+	(void)old;
+}
+
+static void cancel_data_ready(struct bt_l2cap_br_chan *br_chan)
+{
+	struct bt_conn *conn = br_chan->chan.conn;
+
+	sys_slist_find_and_remove(&conn->upper_data_ready,
+				  &br_chan->_pdu_ready);
+
+	atomic_set(&br_chan->_pdu_ready_lock, 0);
+}
+
 int bt_l2cap_br_send_cb(struct bt_conn *conn, uint16_t cid, struct net_buf *buf,
 			bt_conn_tx_cb_t cb, void *user_data)
 {
 	struct bt_l2cap_hdr *hdr;
+	struct bt_l2cap_chan *ch = bt_l2cap_br_lookup_tx_cid(conn, cid);
```

Comment:
Can `ch` be marked as `const`? Looks like it's not modified in this function

Reply:
sure

Reply:
I tried, CI broke. I can make a follow-up PR to constify everything.

Reply:
If it's non-trivial to add `const`, you can just omit it. As previously mentioned, it's not a big deal for local variables